File: backend_python/api/orders/repository/localdb.py
from .interface import OrdersCollectionBlueprint


_db = {
    "stock": {
        "last_updated": "2024-09-10 12:00:00",
        "beers": [
            {
                "name": "Corona",
                "price": 115,
                "quantity": 2
            },
            {
                "name": "Quilmes",
                "price": 120,
                "quantity": 0
            },
            {
                "name": "Club Colombia",
                "price": 110,
                "quantity": 3
            }
        ]
    },
  "orders": {
    "bcdf1d77-bdf8-49de-8697-570380d40334": {
    "_id": "bcdf1d77-bdf8-49de-8697-570380d40334",
    "created": "2024-09-10 12:00:00",
    "paid": False,
    "subtotal": 0,
    "taxes": 0,
    "discounts": 0,
    "items": [],
    "rounds": [
        {
            "created":  "2024-09-10 12:00:30",
            "items": [
                {
                    "name": "Corona",
                    "quantity": 2
                },
                {
                    "name": "Club Colombia",
                    "quantity": 1
                }
            ]
        },
        {
            "created":  "2024-09-10 12:20:31",
            "items": [
                {
                    "name": "Club Colombia",
                    "quantity": 1
                },
                {
                    "name": "Quilmes",
                    "price": 2
                }
            ]
        },
        {
            "created":  "2024-09-10 12:43:21",
            "items": [
                {
                    "name": "Quilmes",
                    "quantity": 3
                }
            ]
        }
      ]
    }
  }
}

class OrdersCollection(OrdersCollectionBlueprint):
    __instance = None
    
    @staticmethod
    def getInstance():
        if not OrdersCollection.__instance:
            OrdersCollection.__instance = OrdersCollection()
        return OrdersCollection.__instance
    
    def find(self):
        return _db

File: backend_golang/api/orders/repository/local_db.go
package repository

import (
	"backend_golang/api/orders"
	"backend_golang/domain/models"
	"context"
)

type localDBRepository struct {
}

// NewLocalDBRepository ...
func NewLocalDBRepository() orders.Repository {
	return &localDBRepository{}
}

func (r *localDBRepository) Find(simpleContext context.Context) (*models.Store, error) {
	return db, nil
}

var db = &models.Store{
	Stock: models.Stock{
		LastUpdated: "2024-09-10 12:00:00",
		Beers: []models.StockBeer{
			{
				Name:     "Corona",
				Price:    115,
				Quantity: 2,
			},
			{
				Name:     "Quilmes",
				Price:    120,
				Quantity: 0,
			},
			{
				Name:     "Club Colombia",
				Price:    110,
				Quantity: 3,
			},
		},
	},
	Orders: map[string]models.Order{
		"bcdf1d77-bdf8-49de-8697-570380d40334": {
			ID:        "bcdf1d77-bdf8-49de-8697-570380d40334",
			Created:   "2024-09-10 12:00:00",
			Paid:      false,
			Subtotal:  0,
			Taxes:     0,
			Discounts: 0,
			Items:     []models.OrderItem{},
			Rounds: []models.OrderRound{
				{
					Created: "2024-09-10 12:00:30",
					Items: []models.RoundItem{
						{
							Name:     "Corona",
							Quantity: 2,
						},
						{
							Name:     "Club Colombia",
							Quantity: 1,
						},
					},
				},
				{
					Created: "2024-09-10 12:20:31",
					Items: []models.RoundItem{
						{
							Name:     "Club Colombia",
							Quantity: 1,
						},
						{
							Name:     "Quilmes",
							Quantity: 2,
						},
					},
				},
				{
					Created: "2024-09-10 12:43:21",
					Items: []models.RoundItem{
						{
							Name:     "Quilmes",
							Quantity: 3,
						},
					},
				},
			},
		},
	},
}
